### DISCUSSION TITLE
How to change footer for target page number?

### DISCUSSION THREAD

**Discussion by Cheshirre** [QuestPDF/QuestPDF#45] [Q&A]
I have a document with 3 pages. There are two types of footer. 
First type for 1st and 3rd pages. And the second type for second page.
How I can achieve this goal?

**Comment by MarcinZiabek** [comment]
Hi, are those pages somehow connected, e.g. by common pageable content? Or each page is like a separate entity? Can you please provide an example?

**Reply by Cheshirre** [reply]
The same header for all pages. 
And some reusable components on this pages. 
It is one class implements IDocument with page in compose. 
I will attach an example tomorrow.

**Reply by MarcinZiabek** [reply]
Sure :) I am asking for more details, as the answer depends on your needs :) We already have a couple of ways to achieve described requirements. However, the most generic and useful one is still in the experimental phase.

You may try to use multiple `ShowOnce` and `SkipOnce` elements to drive proper behaviour. https://www.questpdf.com/documentation/api-reference.html#show-once

Or, using multiple page types: https://www.questpdf.com/documentation/patterns-and-practices.html#page-settings

**Reply by Cheshirre** [reply]
public void Compose(IDocumentContainer container)
        {
            container
                .Page(page =>
                {
                    page.Margin(50);

                    page.Header().Element(ComposeHeader);
                    page.Content().Element(ComposeContent);

                    page.Footer().AlignRight().Stack( stack =>
                        {
                            stack.Item().Text(x =>
                            {
                                var textStyle = TextStyle.Default.Size( 9 ).FontType("Times New Roman");

                                x.Span( "Footer for all other pages", textStyle );
                            });
                            stack.Item().Text(x =>
                            {
                                var textStyle = TextStyle.Default.Size( 9 ).FontType("Times New Roman");
                            
                                x.Span( "Footer only for second page", textStyle );
                            });
                        } );
                });
        }

**Reply by Cheshirre** [reply]
Is there any settings like width alignment in MS Word? For all text in paragraph, for example.

**Reply by Cheshirre** [reply]
Content on first page is dynamic and continues on second page.

**Comment by MarcinZiabek** [comment]
Please try this:

```csharp
container.Page(page =>
{
    page.Margin(50);
    page.Background(Colors.White);
    
    // placeholders
    page.Content().Stack(stack =>
    {
        stack.Spacing(20);
        
        foreach (var _ in Enumerable.Range(1, 50))
        {
            stack
                .Item()
                .Background(Colors.Grey.Lighten2)
                .Height(50);
        }
    });
    
    page.Footer().Stack(stack =>
    {
        // 1st page
        stack.Item().ShowOnce().Background(Colors.Red.Lighten2).Height(100);
        
        // 2nd page
        stack.Item().SkipOnce().ShowOnce().Background(Colors.Orange.Lighten2).Height(100);
        
        // 2rd page and all following
        stack.Item().SkipOnce().SkipOnce().Background(Colors.Green.Lighten2).Height(100);
    });
});
```

Future versions of the library may provide more tools regarding such layouts.

**Reply by MarcinZiabek** [reply]
@Cheshirre Please try the solution above and confirm if it meets your requirements :) If a more complex mechanism is needed, please let me know and we can work together to make it happen in one of the future releases.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
